atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraGetRowKeysEteTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.ete;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.rows.GetRowKeysResource;
+import com.palantir.common.random.RandomBytes;
+
+public class CassandraGetRowKeysEteTest {
+    private GetRowKeysResource client = EteSetup.createClientToSingleNode(GetRowKeysResource.class);
+
+    @Before
+    public void setup() {
+        client.resetTable();
+    }
+
+    @Test
+    public void canGetRowKeysOnEmptyTable() {
+        assertThat(client.getRowKeys(0, 10)).isEmpty();
+    }
+
+    @Test
+    public void getRowKeysReturnsOneResultPerRowStartingWithStartRow() {
+        generateCellsWithNumberedRows(3).forEach(client::insertCell);
+        generateCellsWithNumberedRows(5).forEach(client::insertCell);
+        generateCellsWithNumberedRows(4).forEach(client::insertCell);
+        assertListsContainsExactlyRowsFromTo(client.getRowKeys(1, 10), 1, 5);
```

Comment:
Move to batch size of 5? If this was broken, we're currently only 1 result off from not returning 5 given the number of results per row key.